pandas/_libs/reduction.pyx

```diff
@@ -158,7 +158,7 @@ cdef class _BaseGrouper:
         if util.is_array(values) and not values.flags.contiguous:
             # e.g. Categorical has no `flags` attribute
             values = values.copy()
-        index = dummy.index.values
+        index = dummy.index.to_numpy()
```

Comment:
for EA cases, we want to avoid going through libreduction altogether

Reply:
What do you mean? We shouldn't be hitting this code path in the first place?

Reply:
hmm, yeah what does this do to the categorical grouping case. This should be a highly optimized path, is this still?

Reply:
> What do you mean? We shouldn't be hitting this code path in the first place?

Correct.  This is called from core.apply.FrameApply.apply_standard after a check:

```
        if (
            self.result_type in ["reduce", None]
            and not self.dtypes.apply(is_extension_array_dtype).any()
            # Disallow complex_internals since libreduction shortcut
            #  cannot handle MultiIndex
            and not isinstance(self.agg_axis, ABCMultiIndex)
        ):
```

The MultiIndex check may need to be expanded to include EAs.

I think this is called from one other place in non-test code, so the missing check may belong there.

Reply:
I vaguely recall a `_has_complex_...` property that indicated with the Index was backed by an ndarray. Was that removed?

Reply:
`self._has_complex_internals` was equivalent to to `isinstance(self, MultiIndex)`

Reply:
The issue in question doesn't actually go through here, but rather `_aggregate_series_fast`, which dispatches through `libreduction`, but the point still applies that we want to avoid EA backed indexes here, so I've modified the check to use `_has_complex_internals`.